mautic/mautic
# M3 replacing common subscriber

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8094
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Removing dependencies on CommonSubscriber which uses MauticFactory. Subscribes should use DI instead.

I used this PHPSTAN check to ensure all subscribers has defined classes and properties:
`$ ~/.composer/vendor/phpstan/phpstan-shim/phpstan.phar analyse app/bundles/*Bundle/EventListener`
It uncovered bugs that I could not catch testing manually.

#### Steps to test this PR:
This PR touches pretty much each and every Mauitic feature. I'll try to point out the big ones here.
1. Load up [this PR](https://mautibox.com)
2. Make sure that Mautic configuration works as before.
3. Make sure search works as before.
4. Test campaigns.
5. Test Dashboard widgets.
6. Test import.

#### List deprecations along with the new alternative:
See https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

#### List backwards compatibility breaks:
See https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#
